cedar-dafny/validation/typechecker.dfy

```diff
@@ -381,75 +389,98 @@ module validation.typechecker {
       }
     }
 
+    function extractEntityType(lub: EntityLUB): Result<EntityType> {
+      match lub {
+        case AnyEntity => Result.Err(NonSingletonLub)
+        case EntityLUB(tys) =>
+          def.util.EntityTypeLeqIsTotalOrder();
+          var tySeq := def.util.SetToSortedSeq(lub.tys,def.util.EntityTypeLeq);
+          if |tySeq| == 1
+          then Result.Ok(tySeq[0])
+          else Result.Err(NonSingletonLub)
```

Comment:
I don't think the `NonSingletonLub` error can ever be raised -- shouldn't strict validation have returned an error instead of constructing a non-singleton entity LUB?

Reply:
The context might contain a non-singleton LUB. It'd probably be easy enough to prove this error never occurs given an strictly-typed request environment.